core/src/main/java/org/elasticsearch/index/query/TermsQueryParser.java

```diff
@@ -101,13 +80,14 @@ public Query parse(QueryShardContext context) throws IOException, QueryParsingEx
                     throw new QueryParsingException(parseContext, "[terms] query does not support multiple fields");
                 }
                 fieldName = currentFieldName;
-
+                
+                values = Lists.newArrayList();
```

Comment:
because of the way the constructors are setup we can never have values set to null, but rather values set to an empty list.